### PR TITLE
feat: Least Access Privilege iam BYOB

### PR DESCRIPTION
**Pull request by wandb-zacharyblasczyk**
This pull requests addresses issues customers have with our account and the permissions it has when they bring their own bucket and kms key.

It specifically addresses:
- kms
- s3

### CODE REVIEW

File: examples/byob/main.tf [hunk 1]
```diff
@@ -87,7 +93,17 @@ resource "aws_s3_bucket_policy" "default" {
...
+        "Action" : [
+          "s3:GetObject*",
+          "s3:GetEncryptionConfiguration",
+          "s3:ListBucket",
+          "s3:ListBucketMultipartUploads",
+          "s3:ListBucketVersions",
+          "s3:AbortMultipartUpload",
+          "s3:DeleteObject",
+          "s3:PutObject",
+          "s3:PutObjectAcl"
```
Comment (vanpelt) [inline examples/byob/main.tf:105]: I think we can get rid of this permission.  It's only used to make objects public and in private cloud objects should almost never be public.  I can imagine a use-case for it but let's get rid of it until a user asks for it.

**Review by vanpelt — DISMISSED** [review]

File: examples/byob/main.tf (hunk 1 above)
Comment (wandb-zacharyblasczyk) [reply examples/byob/main.tf:105]: I see what you are saying. Just line 105?

Do you want it behind a feature flag called `enable_public-reports` or something?

That is the only use case I can think of.

++ @venky-wandb

Comment (vanpelt) [reply examples/byob/main.tf:105]: Lets just get rid of it.  I'm pretty sure we disable the creation of public objects at the bucket level anyway which most enterprises require.

Comment (elainaRenee) [reply examples/byob/main.tf]: I'm planning on adding some better in-app messaging for this case as well (i.e. user tries to make an object public when the bucket doesn't allow public ACLs)

Comment (wandb-zacharyblasczyk) [reply examples/byob/main.tf]: Resolved.

File: examples/byob/main.tf [hunk 2]
```diff
@@ -87,7 +93,16 @@ resource "aws_s3_bucket_policy" "default" {
...
-        "Action" : "s3:*",
+        "Action" : [
+          "s3:GetObject*",
+          "s3:GetEncryptionConfiguration",
+          "s3:ListBucket",
+          "s3:ListBucketMultipartUploads",
+          "s3:ListBucketVersions",
+          "s3:AbortMultipartUpload",
+          "s3:DeleteObject",
+          "s3:PutObject"
```
Comment (elainaRenee) [inline examples/byob/main.tf:104]: Can we add these ones too?

s3:GetBucketCORS
s3:GetBucketLocation
s3:GetBucketVersioning

These are used for validating the bucket has the right config

Comment (wandb-zacharyblasczyk) [reply examples/byob/main.tf:104]: Absolutely! Sorry for the delay.

**Review by elainaRenee — APPROVED** [review]
lgtm

**Comment by jsbroks** [comment]
This PR is included in version 1.9.0 :tada:

**Pull request merged**